records/generator.go

```diff
@@ -207,10 +207,10 @@ func (rg *RecordGenerator) findMaster(masters ...string) (state.State, error) {
 			logging.Error.Println(err)
 		}
 
-		if sj, err = rg.loadWrap(ip, port); err == nil && sj.Leader != "" {
+		if sj, err = rg.loadWrap(credentials, ip, port); err == nil && sj.Leader != "" {
 			return sj, nil
 		}
-		logging.Verbose.Println("Warning: Zookeeper is wrong about leader")
+		logging.Verbose.Println("Warning: Zookeeper is wrong about leader, or request failed")
```

Comment:
I modified this warning message because I was seeing it when Zookeeper was right about the leader, but authentication was failing.